android/src/main/kotlin/com/example/user_location_plugin/UserLocationPlugin.kt

```diff
@@ -1,35 +1,143 @@
+        permissionEventChannel =
+            EventChannel(flutterPluginBinding.binaryMessenger, PERMISSION_EVENT_CHANNEL)
+        permissionEventChannel.setStreamHandler(permissionStreamHandler)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            GET_PLATFORM_VERSION -> getPlatformVersion(result)
+            REQUEST_PERMISSION -> requestPermission(result)
+            CHECK_PERMISSION -> checkPermission(result)
+            else -> result.notImplemented()
+        }
+    }
+
+
+    private fun requestPermission(result: Result) {
+        context?.run {
+            activity?.apply {
+                requestPermissions(
+                    this,
+                    arrayOf(
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    ),
+                    REQUEST_PERMISSION_CODE
+                )
+            }
+        }
+        result.success(null)
+    }
+
+    private fun checkPermission(result: Result): Boolean {
+        var permitted = false
+        context?.run {
+            permitted = ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+                    && ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+            permissionEventSource?.success(
+                if (permitted) PERMISSION_GRANTED else permissionEventSource?.success(
+                    PERMISSION_DENIED
+                )
+            )
+        }
+        result.success(permitted)
+        return permitted
+    }
+
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activity = binding.activity
+        binding.addActivityResultListener(this)
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        Log.i("onDetachedConfigChanges", "onDetachedFromConfigChanges")
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        Log.i("onReattachedToActivity", "onReattachedToActivityForConfigChanges")
+    }
+
+    override fun onDetachedFromActivity() {
+        Log.i("onDetachedFromActivty", "onDetachedFromActivity")
+    }
+
+    private fun getPlatformVersion(result: Result) {
+        result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
+        Log.i("onActivityResult", "onActivityResult")
+        return true
+    }
+
+    companion object PluginConstants {
```

Comment:
Is not necessary to name the companion object, because you know if you use a companion the variables that you define only you are using in this file

Reply:
name removed